docs/source/x_ext.rst

```diff
@@ -557,32 +560,42 @@ Commit interface
+  |                    |                        | Shall an offloaded instruction be commited or killed?                                                                        |
+  |                    |                        | If ``commit_valid`` is 1 and ``commit_kill`` is 0,  then the core guarantees that the offloaded instruction (``id``) and any |
+  |                    |                        | older (i.e. preceding) instructions are no longer speculative, will not get killed (e.g. due to misspeculation or an        |
+  |                    |                        | exception in a preceding instruction), and are allowed to be committed.                                                      |
+  |                    |                        | If ``commit_valid`` is 1 and ``commit_kill`` is 1, then the offloaded instruction (``id``) and any newer (i.e. succeeding)   |
+  |                    |                        | instructions shall be killed in the |coprocessor| and the |coprocessor| must guarantee that the related instructions do/did  |
+  |                    |                        | not change architectural state.                                                                                              |
   +--------------------+------------------------+------------------------------------------------------------------------------------------------------------------------------+
 
-The ``commit_valid`` signal will be 1 exactly one ``clk`` cycle for every offloaded instruction by the |coprocessor| (whether accepted or not). The ``id`` value indicates which offloaded
-instruction is allowed to be committed or is supposed to be killed.
+The ``commit_valid`` signal will be 1 exactly one ``clk`` cycle.
+It is not required that a commit transaction is performed for each offloaded instruction individually.
+Instructions can be signalled to be non-speculative or to be killed in batch.
+E.g. signalling the oldest instruction to be killed is equivalent to requesting a flush of the |coprocessor|.
+The first instruction to be considered not-to-be-killed after a commit transaction with ``commit_kill`` as 1,
+is at earliest an instruction with successful issue transaction starting at least one clock cycle later.
+
+.. note::
 
-For each offloaded and accepted instruction the core is guaranteed to (eventually) signal that such an instruction is either no longer speculative and can be committed (``commit_valid`` is 1
-and ``commit_kill`` is 0) or that the instruction must be killed (``commit_valid`` is 1 and ``commit_kill`` is 1). 
+  If an instruction is marked in the |coprocessor| as killed or committed, the |coprocessor| shall ignore any subsequent commit transaction related to that instruction.
 
+A |processor| is required to mark every offloaded and accepted instruction as either killed or non-speculative.
```

Comment:
Looking at the old definition: 
> For each offloaded and accepted instruction the core is guaranteed to (eventually) signal [...]

It was already previously the case, that a CPU only needs to signal commit transactions for offladed AND accepted instructions. I don't see how that changed. 
For example, if a pipeline flush happens, while the CPU is attempting to offload, the instruction currently in issue is considered killed, so no need to kill it again. If an instruction is being issued, and CPU decides it wants to offload a different one, then I also do not think that instruction requires killing. It will be offloaded at a later time.

I agree on the second part of your comment though. The situation is however slightly different, as the commit transaction now targets an entire range of ids. So the copro cannot just ignore the commit transaction, but still needs to interpret it also when the id was not accepted by it. Added a note in 4e741ef

Reply:
I am talking about the following note which seemingly got deleted in this PR, but which I think is important (but needs to be rephrased a little because of the rest of the PR):

![image](https://github.com/openhwgroup/core-v-xif/assets/40633348/4e7590ff-392f-482f-b6b9-e9bf767e84a8)

It is essential that a coprocessor can deal with receiving a commit handshake related to IDs it never accepted or even saw and that the behavior is still well defined. Now that more than one instruction might get accepted/killed, this still needs to be done even though the communicated ID is not handled by a given coprocessor.

Reply:
ok, i got confused about the term no accepted. The scenario you refer to is something that I would call rejected. The spec might be a bit ambigous in the wording. E.g. here it clearly states rejected: 

> A transaction is considered not offloaded/rejected on the positive edge of ``clk`` when ``issue_valid`` and ``issue_ready`` are asserted while ``accept`` is 0.

But in other parts it is called non-accepted. I would suggest to harmonize this towards rejected to remove ambiguity. Then it is clearly different from scenarios where the copro did not respond before the CPU retracted the issue.

Updated in d785f3e

Reply:
no I am a bit more confused. Do you think additional changes are needed? I don't think you need to kill an instruction if it did not get an issue_resp.

Reply:
Such instructions indeed do not need to get killed in a coprocessor that did never see the related issue command. The key is that a coprocessor shall be tolerant against receiving such kill (or commit) transactions.

Reply:
Ok, lets make this explicit.

> A |coprocessor| must be tolerant to any possible ``commit.id``, whether this represents and in-flight instruction or not.

fixed in 969d4c6